### PR TITLE
add useAutomaticKeepAlive hook

### PR DESCRIPTION
**Pull request by DevNico**
See #101

### CODE REVIEW

**Comment by DevNico** [comment]
@rrousselGit What do you think?

**Comment by rrousselGit** [comment]
I'll review this later

File: packages/flutter_hooks/test/use_automatic_keep_alive_test.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            textDirection: TextDirection.ltr,
+            child: DefaultTabController(
+              length: 2,
+              child: TabBarView(
+                children: [
+                  HookBuilder(builder: (context) {
+                    final wantKeepAlive = useState(false);
+                    useAutomaticKeepAlive(wantKeepAlive: wantKeepAlive.value);
+                    useEffect(() {
+                      Future.delayed(
```
Comment (rrousselGit) [inline packages/flutter_hooks/test/use_automatic_keep_alive_test.dart:84]: Don't rely on delays inside tests. 

Could you refactor the test to not do that? :pray:

Comment (DevNico) [reply packages/flutter_hooks/test/use_automatic_keep_alive_test.dart:84]: The delay is matched to the pump duration but I'll try to change that.

Comment (rrousselGit) [reply packages/flutter_hooks/test/use_automatic_keep_alive_test.dart:84]: My suggestion would be to have a ValueNotifier in the test (outside of the widget)

and do useValueListenable(valueNotifier)

Then change the pump into a valueNotifier.value = true;

Comment (DevNico) [reply packages/flutter_hooks/test/use_automatic_keep_alive_test.dart]: Great suggestion, works like a charm.

**Comment by DevNico** [comment]
Should I remove the `!` since your ci seems to be on a newer flutter version?

**Comment by rrousselGit** [comment]
This would require bumping the minimum Flutter SDK.

Maybe add an ignore for now, until the fix can be applied to stable.

**Comment by DevNico** [comment]
friendly ping @rrousselGit

**Comment by kNoAPP** [comment]
🙏 @rrousselGit

**Pull request merged**